### PR TITLE
feat: added trigger for collection property editor, making it dd for mongo

### LINKED ISSUE #23962: [Feature]: Mongo collection field should be a combo dropdown
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

e.g., if I add the movies mock db, I don’t know what collections there are. In the mongo query, the collections field should:
1. be a dropdown with options populated from the schema
2. I should also be able to add my own input e.g., some binding
3. if schema is not available, the dropdown can be empty and I should still be able to enter something (this last point is not ideal ux, but we will iterate on it)

### Why should this be worked on?

easy of query writing

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description
Added a dynamic trigger config to mongo editor json for each command to fetch the collections whenever the mongo action is opened.  Below are the properties that are updated :
- Changed the control type for mongo collection to `DROP_DOWN`
- Added `propertyName` to store collections list in redux store with key `get_collections` inside triggers for specific action id
- Added `fetchOptionsConditionally` boolean to true which helps in fetching the list of collections
- Added `alternateViewTypes` to add toggle button `JS` to add text instead of dropdown
- Added ` "conditionals": { "fetchDynamicValues": { "condition": "{{true}}", "config": { "params": { "requestType": "_GET_STRUCTURE", "displayType": "DROP_DOWN" } } } }` to request trigger api for the respective collection list.

#### PR fixes following issue(s)
Fixes #23962

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-plugins/mongoPlugin/src/main/resources/editor/count.json [hunk 1]
```diff
@@ -12,8 +12,22 @@
...
+          "evaluationSubstitutionType": "TEMPLATE",
+          "propertyName": "get_collections",
+          "fetchOptionsConditionally": true,
+          "alternateViewTypes": ["json"],
+          "conditionals": {
+            "fetchDynamicValues": {
+              "condition": "{{true}}",
+              "config": {
+                "params": {
+                  "requestType": "_GET_STRUCTURE",
```
Comment (sneha122) [inline app/server/appsmith-plugins/mongoPlugin/src/main/resources/editor/count.json:25]: @AmanAgarwal041 What does this requestType denote? Do we call getStructure API based on this when we create mongo query?

Comment (AmanAgarwal041) [reply app/server/appsmith-plugins/mongoPlugin/src/main/resources/editor/count.json:25]: Without this `requestType` the trigger api was throwing error (400), so I noticed that there was another Method Itentifier being used for `GoogleSheetPlugin`. I passed this and it started working. I can see the trigger request being handled for each datasource on backend.

Comment (sneha122) [reply app/server/appsmith-plugins/mongoPlugin/src/main/resources/editor/count.json:25]: Cool thanks @AmanAgarwal041

**Review by sneha122 — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/FormNativeToRawTests/MongoConversion_spec.ts [hunk 2]
```diff
@@ -7,10 +7,20 @@ describe("Mongo Form to Native conversion works", () => {
 
     _.dataSources.CreateDataSource("Mongo", true, true);
     _.dataSources.CreateQueryAfterDSSaved();
-    _.agHelper.TypeDynamicInputValueNValidate(
-      "listingAndReviews",
-      formControls.mongoCollection,
+    cy.wait("@trigger");
+    _.agHelper.GetNClick(
+      "[data-testid='t--actionConfiguration.formData.collection.data-JS']",
     );
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/FormNativeToRawTests/MongoConversion_spec.ts:13]: What element is this, why is this needed?

File: app/client/cypress/e2e/Regression/ClientSide/FormNativeToRawTests/MongoConversion_spec.ts [hunk 3]
```diff
@@ -7,10 +7,20 @@ describe("Mongo Form to Native conversion works", () => {
...
+    _.agHelper.EnterValue("listingAndReviews", {
+      propFieldName: "",
+      directInput: false,
+      inputFieldName: "Collection",
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/FormNativeToRawTests/MongoConversion_spec.ts:17]: If its dropdown, why is EnterValue?

File: app/client/cypress/e2e/Regression/ServerSide/QueryPane/EvaluatedValuePopUp_spec.ts [hunk 4]
```diff
@@ -6,9 +6,12 @@ describe("Ensures evaluated popup is viewable when dynamic bindings are present
     _.dataSources.CreateDataSource("Mongo", true, true);
     _.dataSources.CreateQueryAfterDSSaved();
     // ordinary strings should not open evaluated value popup
-    _.agHelper.TypeDynamicInputValueNValidate(
+    cy.wait("@trigger");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ServerSide/QueryPane/EvaluatedValuePopUp_spec.ts:9]: What is this wait for?

File: app/client/cypress/e2e/Regression/ServerSide/QueryPane/Mongo_Spec.ts [hunk 5]
```diff
@@ -730,13 +735,18 @@ describe("Validate Mongo Query Pane Validations", () => {
...
     _.dataSources.ValidateNSelectDropdown(
       "Commands",
       "Find document(s)",
       "Insert document(s)",
     );
 
     _.agHelper.RenameWithInPane("InsertBirthNDeath");
+    _.agHelper.GetNClick(
+      "[data-testid='t--actionConfiguration.formData.collection.data-JS']",
+    );
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ServerSide/QueryPane/Mongo_Spec.ts:749]: No direct locator in spec.

File: app/client/cypress/e2e/Regression/ClientSide/FormNativeToRawTests/MongoConversion_spec.ts (hunk 2 above)
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/FormNativeToRawTests/MongoConversion_spec.ts:13]: It is switchable input. Dropdown can be converted to input text

File: app/client/cypress/e2e/Regression/ClientSide/FormNativeToRawTests/MongoConversion_spec.ts (hunk 3 above)
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/FormNativeToRawTests/MongoConversion_spec.ts:17]: It is both. Switchable input. Dropdown to json input.

File: app/client/cypress/e2e/Regression/ServerSide/QueryPane/EvaluatedValuePopUp_spec.ts (hunk 4 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ServerSide/QueryPane/EvaluatedValuePopUp_spec.ts:9]: Use `agHelper.AssertNetworkStatus`. not simply wait.

Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ServerSide/QueryPane/EvaluatedValuePopUp_spec.ts:9]: When the input is rendered it has a property of fetching dynamic option values. This wait is for the same thing i.e. when the fetching finishes then it would move to another command

File: app/client/cypress/e2e/Regression/ClientSide/FormNativeToRawTests/MongoConversion_spec.ts (hunk 3 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/FormNativeToRawTests/MongoConversion_spec.ts:17]: Need snap of same please to better understand.

File: app/client/cypress/e2e/Regression/ServerSide/QueryPane/Mongo_Spec.ts (hunk 5 above)
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ServerSide/QueryPane/Mongo_Spec.ts]: fixed

File: app/client/cypress/e2e/Regression/ClientSide/FormNativeToRawTests/MongoConversion_spec.ts (hunk 2 above)
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/FormNativeToRawTests/MongoConversion_spec.ts]: https://github.com/appsmithorg/appsmith/pull/24213#discussion_r1227812704

File: app/client/cypress/e2e/Regression/ServerSide/QueryPane/EvaluatedValuePopUp_spec.ts (hunk 4 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ServerSide/QueryPane/EvaluatedValuePopUp_spec.ts:9]: Please use `agHelper.AssertNetworkStatus` instead of `cy.wait("@trigger");`

File: app/client/cypress/e2e/Regression/ServerSide/QueryPane/Mongo_Spec.ts [hunk 6]
```diff
@@ -268,16 +269,18 @@ describe("Validate Mongo Query Pane Validations", () => {
...
       "Find document(s)",
       "Insert document(s)",
     );
 
-    _.agHelper.EnterValue("AuthorNAwards", {
-      propFieldName: "",
-      directInput: false,
-      inputFieldName: "Collection",
+    _.dataSources.updateJSFieldWithInputValue({
+      fieldProperty: formControls.mongoCollectionPath,
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ServerSide/QueryPane/Mongo_Spec.ts:281]: Create this locator in Datasources.ts or check if an locator already eists for this in same file & use. 
For this single locator - `import formControls from "../../../../locators/FormControl.json"; `is not needed.

Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ServerSide/QueryPane/Mongo_Spec.ts:281]: It is being used at multiple places

File: app/client/cypress/support/Pages/DataSources.ts [hunk 7]
```diff
@@ -204,6 +204,7 @@ export class DataSources {
   public _datasourceModalDoNotSave = ".t--datasource-modal-do-not-save";
   public _cancelEditDatasourceButton = ".t--cancel-edit-datasource";
   public _urlInputControl = "input[name='url']";
+  public mongoCollectionPath = "t--actionConfiguration.formData.collection";
```
[the remark this answers is not in the document]
Comment (AmanAgarwal041) [reply app/client/cypress/support/Pages/DataSources.ts:207]: Its not a locator, its a property path to refer to the `data-testid="t--actionConfiguration.formData.collection.data-JS"` button for switching to JSON

File: app/client/cypress/e2e/Regression/ServerSide/QueryPane/Mongo_Spec.ts (hunk 6 above)
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ServerSide/QueryPane/Mongo_Spec.ts]: `updateJSFieldWithInputValue` is made generic for properties having Switch to JSON functionality, if the fieldProperty is passed, then it will refer to that particular field's switch to json button.

File: app/client/cypress/e2e/Regression/ClientSide/IDE/MaintainContext&Focus_spec.js [hunk 8]
```diff
@@ -1,5 +1,6 @@
 import reconnectDatasourceModal from "../../../../locators/ReconnectLocators";
 import * as _ from "../../../../support/Objects/ObjectsCore";
+import formControls from "../../../../locators/FormControl.json";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/IDE/MaintainContext&Focus_spec.js]: Is this import needed?

File: app/client/cypress/e2e/Regression/ServerSide/QueryPane/Mongo_Spec.ts [hunk 9]
```diff
@@ -1,5 +1,6 @@
 import { INTERCEPT } from "../../../../fixtures/variables";
 import * as _ from "../../../../support/Objects/ObjectsCore";
+import formControls from "../../../../locators/FormControl.json";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ServerSide/QueryPane/Mongo_Spec.ts]: This needed??

Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ServerSide/QueryPane/Mongo_Spec.ts]: removed

File: app/client/cypress/e2e/Regression/ClientSide/IDE/MaintainContext&Focus_spec.js (hunk 8 above)
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/IDE/MaintainContext&Focus_spec.js]: removed

**Review by sneha122 — APPROVED** [review]
